### PR TITLE
Preventing cutting unicode characters in half

### PR DESCRIPTION
**Pull request by shiroyasha**
Now, that we are no longer waiting for the newline character to flush the output buffer, a new bug got introduced. We are cutting Unicode characters in-half.

Reminder. A UTF-8 character can span to up to 4 bytes. The highest bit in the (8th bit) marks if the character is finished or there is more upcoming information in the next byte.

Example:

A UTF-8 character spanning 3 bytes:

10101010 -> 10101010 -> 00101010

A UTF-8 character spanning 1 byte:

00101010

### CODE REVIEW

**Comment by shiroyasha** [comment]
@DamjanBecirovic @radwo @bmarkons this is a pretty hard bug to fix. So far, this is my best idea. I would appreciate your review.

File: pkg/shell/process.go [hunk 1]
```diff
@@ -74,16 +74,61 @@ func (p *Process) OnStdout(callback func(string)) {
...
+			// The unicode continuation sign is marked by the highest (8th) bit.
+			// If the bit is set, it means that the unicode character is not yet
+			// finished.
+			//
+			// In the bellow loop, we are cutting of the last 3 charactes in case
+			// they are marked as the unicode continuation characters.
+			//
+			// An unicode sequence can't be longer than 4 bytes
+			//
+			unicodeContinuationMask := uint(1 << 7)
```
Comment (dnozay) [inline pkg/shell/process.go]: https://en.wikipedia.org/wiki/UTF-8
bug: continuation bytes start with `10xxxxxx`; `1xxxxxxx` also include leading bytes.

Comment (dnozay) [reply pkg/shell/process.go]: This happens very frequently when doing box drawing
```
┏┳━━━━━━━━━━━━━���━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━���━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━���━━
┃┃ TEST Report
┗┻━━━━━━━━━━━━━━━━━━━━���━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━���━━━━━━━━━━━━━━━━━━━━━━━━━━━
```
Box drawing characters are 3 bytes.

Comment (shiroyasha) [reply pkg/shell/process.go]: That's it! Thanks for finding the bug and sharing the solution. 🙇

Comment (shiroyasha) [reply pkg/shell/process.go]: Work on the fix started here https://github.com/semaphoreci/agent/pull/79.

Comment (shiroyasha) [reply pkg/shell/process.go]: @dnozay I started with the implementation of the patch, but unfortunately, I had a hard time replicating the issue in the Agent's tests.

My current assumption is that the root of the problem is not directly in the Agent, but upstream in the log processing service. We are investigating further.

I'll update as soon as we find something.

**Pull request merged**